index.js

```diff
@@ -83,25 +98,34 @@ function createWrapFn(originalsToProxies, proxiesToOriginals, unwrapFn) {
                 // in case when private symbols is exposed via some part of public API
                 // we have to add such symbol to all possible targets where it could appear
                 if (typeof retval === 'symbol' /* && retval.private */) {
-                    allHandlers.forEach(handler => handlePrivate(handler, retval));
+                    // TODO
+                    // allHandlers.forEach(handler => handlePrivate(handler, retval));
                 }
 
-                return wrap(retval);
+                return unwrap(retval);
             },
             get(target, p, receiver) {
-                receiver = unwrapFn(receiver);
+                receiver = unwrap(receiver);
                 //       but we use `original` here instead of `target`
                 //                         ↓↓↓↓↓↓↓↓
                 const retval = Reflect.get(original, p, receiver);
 
                 // in case when private symbols is exposed via some part of public API
                 // we have to add such symbol to all possible targets where it could appear
                 if (typeof retval === 'symbol' /* && retval.private */) {
-                    allHandlers.forEach(handler => handlePrivate(handler, retval));
+                    // TODO
+                    // allHandlers.forEach(handler => handlePrivate(handler, retval));
                 }
 
-                return wrap(retval);
+                return unwrap(retval);
             },
+            set(target, p, value, receiver) {
+                value = unwrap(value);
+                receiver = unwrap(receiver);
+
+                return Reflect.set(target, p, value, receiver);

```

Comment:
This line seems to be not very correct.
You probably wanted `original` instead of `target`, since last one here is `privateHandler` not an object.

Reply:
Note that fixing this and uncommenting everything you marked as `// TODO` leads to all tests passing.

Reply:
```suggestion
                return Reflect.set(original, p, value, receiver);
```